code/game/machinery/adv_med.dm

```diff
@@ -255,8 +255,8 @@
 	if(src.occupant)
 		user << "<span class='notice'>\The [src] is already occupied!</span>"
 		return
-	if(G.affecting.buckled)
-		G.affecting.buckled.unbuckle()
+	if(G.affecting.locked_to)
+		G.affecting.locked_to.unlock_atom(G.affecting)
```

Comment:
It might be less confusing in these cases if you made a simpler proc for this behaviour.
Something like unlock_from()